tests/deployments/databricks/test_databricks.py

```diff
@@ -46,6 +46,38 @@ def test_create_endpoint():
         assert resp == {"name": "test"}
 
 
+def test_create_endpoint_config_only():
```

Comment:
Can we add tests for the error handling logic that needs to be implemented for backwards compatibility / validation of the deprecated state? 
We need to verify that setting `name` in both `config` and as a named arg throws. 
We need to verify that setting `route_optimized` in both will throw.
We need to ensure that the correct warning is emitted when both are provided but the values match. 
We need to ensure that a deprecation warning is emitted when named args are used and they are not set in the config (separate warning from the previous row)